src/simulate/event_transformer/Transformer.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import utils
+import random
+import json
+
+
+class Transformer:
+    CONST_USER_STR = "https://mcommunity.umich.edu/#profile:"
+
+    def __init__(self, eventData, config_data):
+        self.eventData = eventData
+        self.envelope = eventData
+        self.event = self.eventData['data'][0]
+        self.config_data = config_data
+
+    def change_user_in_event(self):
+        try:
+            sample_users = self.config_data['users']
+        except KeyError:
+            logging.error("configuration yaml is missing the \"users\" key")
+            return
+
+        if sample_users is None:
+            logging.error("No Users information added to the configuration yml file")
+            return
+        user_id = self.CONST_USER_STR + random.choice(sample_users)['id']
+        self.event['actor']['id'] = user_id
+        self.event['session']['user']['id'] = user_id
+        self.event['membership']['member'] = user_id
+
+    def change_course_info(self):
+        action = self.event['action']
+        try:
+            sample_courses = self.config_data['courses']
+        except KeyError:
+            logging.error("configuration yaml is missing the \"courses\" key")
+            return
+
+        if sample_courses is None:
+            logging.error("No courses information added to the configuration yml file")
+            return
+
+        a_course_info = random.choice(sample_courses)
+        # targeted to Media events
+        if action != 'LoggedIn':
+            logging.debug("Action=%s", action)
+            object = self.event['object']
+            target = self.event['target']
+            video_player = random.choice(a_course_info['video_player'])
+            object['id'] = video_player
+            object['isPartOf']['id'] = a_course_info['course_site']
+            object['isPartOf']['name'] = a_course_info['course_name']
+            target['id'] = video_player
```

Comment:
`target['currentTime']` should also be given a random value.  It needs to be less than `object['duration']`.

Reply:
Done